osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -69,7 +69,12 @@ public abstract class ScoreProcessor
         /// <summary>
         /// Whether the score is in a failed state.
         /// </summary>
-        public virtual bool HasFailed => Health.Value == Health.MinValue;
+        public virtual bool HasFailed => alreadyFailed;

```

Comment:
Think we can remove `alreadyFailed`, and just keep `HasFailed` as a public-get-private-set property?